client/state/selectors/test/get-selected-or-primary-site-id.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedOrPrimarySiteId } from '../';
+
+describe( 'getSelectedOrPrimarySiteId()', () => {
+	context( 'with no site selected', () => {
+		it( 'should return null if there is no current user', ( ) => {
+			const siteId = getSelectedOrPrimarySiteId( {
+				currentUser: {},
+				ui: {},
+			} );
+			expect( siteId ).to.be.null;
+		} );
+
+		it( 'should return current user\'s primary site\'s ID', () => {
+			const siteId = getSelectedOrPrimarySiteId( {
+				currentUser: { id: 12345678 },
+				ui: {},
+				users: { items: { 12345678: { primary_blog: 7654321 } } },
+			} );
+			expect( siteId ).to.equal( 7654321 );
+		} );
+	} );
+
+	context( 'with a site selected', () => {
+		it( 'should return the selected site\'s ID', ( ) => {
```

Comment:
Yeah, I was considering that too, but I wasn't sure how to group tests for that scenario. Maybe I'll just add a current user and primary site to this test case... :thinking: